app/app.go

```diff
@@ -229,6 +241,33 @@ func NewOsmosisApp(
 	wasmOpts []wasmkeeper.Option,
 	baseAppOptions ...func(*baseapp.BaseApp),
 ) *OsmosisApp {
+
+	if isOTELEnabled {
+		// TODO: handle proper cancellation on shutdown.
+		ctx := context.Background()
+
+		// resource.WithContainer() adds container.id which the agent will leverage to fetch container tags via the tagger.
+		res, err := resource.New(ctx, resource.WithContainer(),
+			resource.WithAttributes(semconv.ServiceNameKey.String("osmosis")),
```

Comment:
We need to replace osmosis with a variable. The environment variable DD_SERVICE=osmosis-fn does not override it